parser_library/test/debugging/debugger_test.cpp

```diff
@@ -34,6 +36,46 @@ using namespace hlasm_plugin::parser_library::debugging;
 using namespace hlasm_plugin::parser_library::workspaces;
 using namespace hlasm_plugin::utils::resource;
 
+namespace {
+
+struct expected_stack_frame
+{
+    expected_stack_frame(size_t begin_line, size_t end_line, uint32_t id, std::string name, std::string source)
+        : begin_line(begin_line)
+        , end_line(end_line)
+        , id(id)
+        , name(std::move(name))
+        , frame_source(std::move(source))
+    {}
+    expected_stack_frame(size_t begin_line,
+        size_t end_line,
+        uint32_t id,
+        std::string name,
+        std::function<bool(std::string_view)> source_matcher)
+        : begin_line(begin_line)
+        , end_line(end_line)
+        , id(id)
+        , name(std::move(name))
+        , frame_source(std::move(source_matcher))
+    {}
+    size_t begin_line;
+    size_t end_line;
+    uint32_t id;
+    std::string name;
+    std::variant<std::string, std::function<bool(std::string_view)>> frame_source;
+
+    static bool check_frame_source(const std::string& exp, std::string_view uri) { return exp == uri; }
+    static bool check_frame_source(const std::function<bool(std::string_view)>& exp, std::string_view uri)
+    {
+        return exp(uri);
```

Comment:
Maybe `assert`?

```suggestion
        assert(exp);
        return exp(uri);
```

Reply:
if `exp` is empty, it is going to throw something anyway...